packages/appkit/src/client.ts

```diff
@@ -1244,6 +1250,12 @@ export class AppKit {
       }
     })
 
+    adapter?.on('pendingTransactions', () => {
+      if (AccountController.state.address && ChainController.state.activeCaipNetwork?.id) {
+        this.updateBalance()
+      }
+    })
+
```

Comment:
Should we do optional chaining here? I wonder if there is a better way to handle this. We should check if undefined, or assert somewhere earlier in the flow